src/dev/amdgpu/pm4_packet_processor.cc

```diff
@@ -1044,6 +1044,7 @@ PM4PacketProcessor::serialize(CheckpointOut &cp) const
     int num_queues = queues.size();
     Addr id[num_queues];
     Addr mqd_base[num_queues];
+    uint64_t mqd_read_index[num_queues];
```

Comment:
This is fine for now -  I originally added these and you are following the style around it, I understand.  But at some point we should remove these VLAs (variable length arrays) because I don't think they are standard C++.  It happens to be OK with gcc/clang which gem5 supports but I think other compilers like MSVC will error out